src/core/server/core_usage_data/types.ts

```diff
@@ -205,6 +205,7 @@ export interface CoreConfigUsageData {
     };
     apiVersion: string;
     healthCheckDelayMs: number;
+    username: 'none' | 'elastic' | 'kibana' | 'kibana_system' | 'other';
```

Comment:
A couple of questions: 
1) I like that you're reporting `other` for unknown usernames. In the same vein, do we care about the distinction between `elastic`, `kibana`, and `other`? What questions are we trying to answer with this data? If we only care about tracking the deprecated `kibana` user, then perhaps we don't need to report on `elastic`? If this data got into the wrong hands, it could signal that a specific Kibana instance is misconfigured with superuser privileges, which potentially makes it a more attractive target.
2) Now that we support [service account tokens](https://github.com/elastic/kibana/pull/102121), do you think it makes sense to report this usage as well? If so, what do you think about calling this field something slightly more generic (`principal`?), and using this to also report if a service account token is in use? Open to suggestions, but as a starting point:
```ts
    principal: 'none' | 'kibana_system_user' | 'kibana_system_service_account' | ...,
```

Reply:
> 1. I like that you're reporting `other` for unknown usernames. In the same vein, do we care about the distinction between `elastic`, `kibana`, and `other`? What questions are we trying to answer with this data? If we only care about tracking the deprecated `kibana` user, then perhaps we don't need to report on `elastic`?

In #48247, we deprecated using `elasticsearch.username: elastic` in your kibana.yml file, and in #63186 we deprecated the `kibana` user entirely. So I figured we'd be interested in understanding usage of either of these in comparison with the `kibana_system` user, since we eventually want to completely prohibit usage of both the `elastic` and `kibana` users.

> If this data got into the wrong hands, it could signal that a specific Kibana instance is misconfigured with superuser privileges, which potentially makes it a more attractive target.

Usage data doesn't include information like host info; true you can see the cluster ID, but the only place I can think of where that would be useful in identifying a Kibana instance is in our own Cloud environment, wherein Kibana is never configured with the `elastic` user. So I'm not sure if this is a legitimate concern, unless there's something else that I'm missing?

> 2. Now that we support [service account tokens](https://github.com/elastic/kibana/pull/102121), do you think it makes sense to report this usage as well? If so, what do you think about calling this field something slightly more generic (`principal`?), and using this to also report if a service account token is in use? Open to suggestions, but as a starting point:

Sure, that's a great idea. There are other ways of authenticating Kibana to ES though (such as PKI); instead of having a principal of 'none', do you think we should lump everything else into 'other' instead?
Assuming we do still want to identify the elastic user, I'm leaning towards:

```ts
principal: 'elastic_user' | 'kibana_user' | 'kibana_system_user' | 'kibana_system_service_account' | 'other'
```

Reply:
> In #48247, we deprecated using `elasticsearch.username: elastic` in your kibana.yml file, and in #63186 we deprecated the `kibana` user entirely. So I figured we'd be interested in understanding usage of either of these in comparison with the `kibana_system` user, since we eventually want to completely prohibit usage of both the `elastic` and `kibana` users.

Ah I thought that https://github.com/elastic/kibana/pull/48247 only applied to development, I forgot that we logged a warning in production. Thanks for the reminder!

> Usage data doesn't include information like host info; true you can see the cluster ID, but the only place I can think of where that would be useful in identifying a Kibana instance is in our own Cloud environment, wherein Kibana is never configured with the elastic user. So I'm not sure if this is a legitimate concern, unless there's something else that I'm missing?

You're right that we don't include identifiable information. I'm not aware of a legitimate _current_ concern, moreso trying to protect ourselves in case a piece of identifiable information were to accidentally make its way into the payload in the future. That said, I think the risk is so small that this would be useful to an attacker that we can leave it in as you have it.

>  There are other ways of authenticating Kibana to ES though (such as PKI); instead of having a principal of 'none', do you think we should lump everything else into 'other' instead?

I think `none` would still be nice to track -- are there methods other than PKI that you're concerned about? Tracking `none` should allow us to see how many clusters have security enabled, but use a proxy or other external mechanism for authentication to ES

Reply:
> I think `none` would still be nice to track -- are there methods other than PKI that you're concerned about? Tracking `none` should allow us to see how many clusters have security enabled, but use a proxy or other external mechanism for authentication to ES

I guess it's just PKI and external methods like you pointed out!

I suppose it's pedantic but you must have a principal to authenticate, so a principal of "none" doesn't really make sense. If you did use PKI, your x509 certificate's distinguished name would be your principal. It would be nice to detect such usage of PKI but I can't really think of a way to do that.

Sounds like the clearest thing would be to specify 'custom_user' for a custom user, and 'other' for anything else (PKI or external, we can't distinguish). Sound good to you?

Reply:
Sounds good to me!

Reply:
Done in bc1842a!